### PR TITLE
Add initial doc for MTW

### LINKED ISSUE #7144: Document the Account.sol abstraction better
### Expected Behavior

- Account abstraction is well documented
- There is a coherent roadmap/guidance how use cases/applications should think about the abstraction

### Current Behavior

`Accounts.sol` is an account abstraction on top of Ethereum's account model that features some additional features such as:
- Authorizing other keys
- Indicating wallet addresses

However, the abstraction is 1) not well documented and 2) not very consistent. 1) is obvious, but for 2) we are not providing enough guidance how this abstraction impacts use cases like:

- Dapp interaction: I.e. should users interact with the wallet address or account address?
- Recovery: Related to dapp interaction, if its tied to the wallet address, then recovery as specced currently cannot recover funds on dapps
- User display: Currently wallet addresses are displayed which makes sense since a lot of exchanges won't lookup wallet address if you gave them the account address. However, wallet addresses can become "stale" and/or be inconsistent if dapps interact with the account address instead
- wallet interfaces: dappkit and walletconnect likely also do not know the difference, how would this possibly change the API (dappkit right now conveys the wallet address?)

### PR DESCRIPTION
**Pull request by codyborn**
### Description

Add initial documentation for the MTW abstraction. This will be modified once we have 1) an account recovery mechanism 2) an easy mechanism to leverage the MTW in DappKit/WalletConnect.

### Related issues

- Fixes #7144

### CODE REVIEW

File: packages/docs/celo-codebase/protocol/identity/meta-transaction-wallet.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+### Fund Recovery
+
+Meta-transaction wallets can also be useful if a user ever loses their phone and backup account key. Any ERC20 tokens that the EOA has [approved](https://docs.openzeppelin.com/contracts/2.x/api/token/erc20#IERC20-approve-address-uint256-) their meta-transaction wallet to use, can be recovered. In the case of loss of EOA, a new EOA will be created and granted signer rights to the original MTW. This would allow the lost funds to still be accessed, even though the original EOA is unrecoverable. This recovery mechanism hasn't been rolled out yet, but is actively being worked on by cLabs.
+
+## For Dapp Developers
```
Comment (nambrot) [inline packages/docs/celo-codebase/protocol/identity/meta-transaction-wallet.md]: Should we have a section for Wallet developers too?

File: packages/docs/celo-codebase/protocol/identity/meta-transaction-wallet.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Meta-Transaction Wallet
```
Comment (nambrot) [inline packages/docs/celo-codebase/protocol/identity/meta-transaction-wallet.md]: Is the relevant piece for folks the MTW or the fact that there are "account address" vs. "walletAddress"

File: packages/docs/celo-codebase/protocol/identity/meta-transaction-wallet.md (hunk 1 above)
Comment (codyborn) [reply packages/docs/celo-codebase/protocol/identity/meta-transaction-wallet.md]: For this, do you imagine we'd handle things like importing Valora accounts (ie. mnemonic)? We currently don't support Komenci to be used by external wallets but it's something that we're considering as part of Celo Rewards cc @jbsibille

File: packages/docs/celo-codebase/protocol/identity/meta-transaction-wallet.md (hunk 2 above)
Comment (codyborn) [reply packages/docs/celo-codebase/protocol/identity/meta-transaction-wallet.md]: Renaming to "Valora Accounts" to be more clear to readers.

File: packages/docs/celo-codebase/protocol/identity/meta-transaction-wallet.md (hunk 1 above)
Comment (nambrot) [reply packages/docs/celo-codebase/protocol/identity/meta-transaction-wallet.md]: It woudn't even be using Komenci, but if we have an external wallet implementing the attestations protocol, they need to know that they should look up the wallet address? Also they should know about DEK registrations, both for ODIS querying but also tx metadata encryrption?

Comment (nambrot) [reply packages/docs/celo-codebase/protocol/identity/meta-transaction-wallet.md]: It woudn't even be using Komenci, but if we have an external wallet implementing the attestations protocol, they need to know that they should look up the wallet address? Also they should know about DEK registrations, both for ODIS querying but also tx metadata encyrption?

Comment (nambrot) [reply packages/docs/celo-codebase/protocol/identity/meta-transaction-wallet.md]: It woudn't even be using Komenci, but if we have an external wallet implementing the attestations protocol, they need to know that they should look up the wallet address? Also they should know about DEK registrations, both for ODIS querying but also tx metadata encryrption?

File: packages/docs/celo-codebase/protocol/identity/valora-accounts.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# Valora Accounts
+
+Behind every Valora wallet are two types of accounts: an externally owned account (EOA) and a meta-transaction wallet (MTW). EOAs are what most people think of when they imagine a blockchain wallet. EOAs are comprised of an ECDSA public/private key pair. With Valora, this EOA is generated and stored on the user's mobile device and backed up via the mnemonic phrase. A meta-transaction wallet on the other hand is a smart contract that can be used to interact with other smart contracts on behalf of an EOA. In this case you can think of the MTW as a proxy account and the EOA as the only controller of this account.
```
Comment (nategraf) [inline packages/docs/celo-codebase/protocol/identity/valora-accounts.md]:
```suggestion
Behind every Valora wallet are two types of accounts: an externally owned account (EOA) and a meta-transaction wallet (MTW). EOAs are what most people think of when they imagine a blockchain wallet. EOAs are comprised of an ECDSA public/private key pair from which the on-chain address is derived. With Valora, this EOA is generated and stored on the user's mobile device and backed up via the mnemonic phrase. A meta-transaction wallet on the other hand is a smart contract that can be used to interact with other smart contracts on behalf of an EOA. In this case you can think of the MTW as a proxy account and the EOA as the only controller of this account.
```

**Review by nategraf — APPROVED** [review]
This is a great start to documenting our account structure

File: packages/docs/celo-codebase/protocol/identity/meta-transaction-wallet.md (hunk 1 above)
Comment (codyborn) [reply packages/docs/celo-codebase/protocol/identity/meta-transaction-wallet.md]: Good points. Added this to the doc and the CLI as a guide.

**Review by nambrot — APPROVED** [review]

**Pull request merged**